### PR TITLE
Add unwind to populate config

### PR DESCRIPTION
**Pull request by Akrion**
- Adds `unwind` as part of the `populate` config section so it converts the populated values from array of single object to an object.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  We were able to automatically fix some formatting issues in this PR for you!
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/example-types/results.js [hunk 1]
```diff
@@ -81,17 +81,36 @@ let defaults = _.defaults({
...
 
+let rowsToObjectConverter = _.curry((populateConfig, row) => {
+  let singularProps = _.flow(
+    _.pickBy(obj => obj.singularObject),
+    _.keys
+  )(populateConfig)
+
+  return _.flow(
+    _.pick(singularProps),
+    _.mapValues(arr => _.first(arr))
```
Comment (daedalus28) [inline src/example-types/results.js]: `_.first` is curried, so this can be `_.mapValues(_.first)`

File: src/example-types/results.js [hunk 2]
```diff
@@ -81,17 +81,36 @@ let defaults = _.defaults({
   include: [],
 })
 
+let rowsToObjectConverter = _.curry((populateConfig, row) => {
+  let singularProps = _.flow(
+    _.pickBy(obj => obj.singularObject),
```
Comment (daedalus28) [inline src/example-types/results.js]: you can use lodash shorthand as `_.pickBy('singularObject')`

File: test/example-types/results.js [hunk 3]
```diff
@@ -194,4 +196,23 @@ describe('results', () => {
       })
     })
   })
+  describe('rowsToObjectConverter', () => {
+    it('should convert array of objects to a single object based on populate config prop "singularObject"', () => {
+      let results = [
+        { other: { a: 1 }, user: [{ name: 'A' }, { name: 'A' }] },
+        { user: [{ name: 'B' }, { name: 'B' }] },
```
Comment (dubiousdavid) [inline test/example-types/results.js]: These examples seem fairly contrived given the repetition of the `name` value for each element.

File: test/example-types/results.js [hunk 4]
```diff
@@ -194,4 +196,23 @@ describe('results', () => {
...
+      ]
+      let populate = {
+        other: { someProp: 1 },
+        user: { singularObject: true },
```
Comment (dubiousdavid) [inline test/example-types/results.js]: These populate examples seem fairly contrived, given they exclude the normal necessary fields: `schema`, `localField`, and `foreignField`.

File: test/example-types/results.js [hunk 5]
```diff
@@ -194,4 +196,23 @@ describe('results', () => {
...
+      }
+      let converter = rowsToObjectConverter(populate)
+
+      results = _.map(row => _.extend(row, converter(row)), results)
```
[the remark this answers is not in the document]
Comment (Akrion) [reply test/example-types/results.js]: Can you clarify this please? Why do I need to move this? `rowsToObjectConverter` does exactly what it says it does ... converts an array of objects to a singular object based on the existence of `singularObject` in the passed to it `populate` configuration object.

`results = _.map(row => _.extend(row, converter(row)), results)` simply converts each of the rows passed using the `rowsToObjectConverter`.

[the remark this answers is not in the document]
Comment (Akrion) [reply test/example-types/results.js]: Never mind I did not see the `test` prefix here. I thought this was comment in the source vs the test.

**Comment by Akrion** [comment]
Changes made.

File: src/example-types/results.js [hunk 6]
```diff
@@ -81,17 +81,38 @@ let defaults = _.defaults({
   include: [],
 })
 
+let rowsToObjectConverter = _.curry((populateConfig, row) => {
```
Comment (dubiousdavid) [inline src/example-types/results.js]: This should go away in favor of using Mongo's built in `$unwind`.

**Comment by Akrion** [comment]
changes made

**Pull request merged**